ibm/resource_ibm_compute_vm_instance.go

```diff
@@ -194,6 +204,16 @@ func resourceIBMComputeVmInstance() *schema.Resource {
 				Computed: true,
 			},
 
+			"private_security_group_ids": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeInt},
+				Set: func(v interface{}) int {
+					return v.(int)
+				},
+				ForceNew: true,
```

Comment:
Since user's can  add more groups after it is provisioned. I think it should be an updatable field. Same with `public_security_group_ids`.
Only thing is one will need a soft reboot of the VSI in such cases and should be properly documented.

Reply:
I think this might be limited then. Probably we can  add another resource  like 
```
resource ibm_security_group_ifc_binding {
   security_group_id = "${...}"
   virtual_guest_id = "${...}"
   interface_type = "PUBLIC"
}
```
But each addition requires a reboot which I think is not good if we were to add 5 such bindings. This could be used to mitigate the above problem and a more common case would be rules are already created and user would specify them during the VSI placeOrder. 
These are just some of my initial thoughts. Definitely needs some more rounds of it.

Reply:
Additions only require a reboot if you're adding a security group to a network component that has never had security groups on it before.

You can check this by calling https://sldn.softlayer.com/reference/services/SoftLayer_Virtual_Guest_Network_Component/securityGroupsReady

If there is a security group on the network component, and securityGroupsReady returns False, then a reboot is required. Otherwise, any security group application to that network component will take effect without reboot.

Reply:
Awesome 👍, will check this.